sdk/keyvault/keyvault-keys/test/public/list.spec.ts

```diff
@@ -9,239 +9,255 @@ import { assertThrowsAbortError } from "../utils/utils.common";
-    // WARNING: When TEST_MODE equals "record", all of the keys in the indicated KEYVAULT_NAME will be deleted as part of this test.
-    if (!isRecordMode()) {
-      return this.skip();
-    }
-    for await (const properties of client.listPropertiesOfKeys()) {
-      try {
-        await testClient.flushKey(properties.name);
-      } catch {
-        // Nothing to do here.
-      }
-    }
-    for await (const deletedKey of client.listDeletedKeys()) {
-      try {
-        await testClient.purgeKey(deletedKey.name);
-      } catch {
-        // Nothing to do here.
-      }
-    }
-  });
-
-  it("can get the versions of a key", async function() {
-    const keyName = testClient.formatName(`${keyPrefix}-${this!.test!.title}-${keySuffix}`);
-    await client.createKey(keyName, "RSA");
-    let totalVersions = 0;
-    for await (const version of client.listPropertiesOfKeyVersions(keyName)) {
-      assert.equal(
-        version.name,
-        keyName,
-        "Unexpected key name in result from listPropertiesOfKeyVersions()."
+versionsToTest(["7.0", "7.1"]).forEach((serviceVersion) => {
+  const versions = function(versions: SupportedVersions) {
+    return supports(serviceVersion, versions);
+  };
+
+  versions(["7.0", "7.1"]).describe.only(
+    "Keys client - list keys in various ways",
+    async function() {
+      const keyPrefix = `list${env.KEY_NAME || "KeyName"}`;
+      let keySuffix: string;
+      let client: KeyClient;
+      let testClient: TestClient;
+      let recorder: Recorder;
+
+      beforeEach(async function() {
+        const authentication = await authenticate(this, serviceVersion);
+        keySuffix = authentication.keySuffix;
+        client = authentication.client;
+        testClient = authentication.testClient;
+        recorder = authentication.recorder;
+      });
+
+      afterEach(async function() {
+        await recorder.stop();
+      });
+
+      // The tests follow
+
+      // Use this while recording to make sure the target keyvault is clean.
+      // The next tests will produce a more consistent output.
+      // This test is only useful while developing locally.
+      it("can purge all keys", async function(): Promise<void> {
+        // WARNING: When TEST_MODE equals "record", all of the keys in the indicated KEYVAULT_NAME will be deleted as part of this test.
+        if (!isRecordMode()) {
+          return this.skip();
+        }
+        for await (const properties of client.listPropertiesOfKeys()) {
+          try {
+            await testClient.flushKey(properties.name);
+          } catch {
+            // Nothing to do here.
+          }
+        }
+        for await (const deletedKey of client.listDeletedKeys()) {
+          try {
+            await testClient.purgeKey(deletedKey.name);
+          } catch {
+            // Nothing to do here.
+          }
+        }
+      });
+
+      versions(["7.0", "7.1"]).it("can get the versions of a key", async function() {
```

Comment:
What if we had our own describe/it helpers instead of chaining through supports? e.g.

```ts
multiDescribe(["7.0", "7.1"], "Cool service feature", function() {
    multiIt(["7.0"], "can do the thing", async function() { /* test code */ });
}
```

Reply:
I feel this is deviating from the Mocha methods which everyone is familiar with.  Current approach is more like "attributing" Mocha tests, and we still get a lot of existing Mocha features (reporting, inclusive/exclusive execution, etc.) without heavily customizing Mocha.  If time allowed I could explore more into Mocha  internals to see if anything can be done better.  Good thing is that this is testing and we only have KeyVault to support now, there are more room to do breaking changes.

Reply:
I do wish they made it easier to extend their internals, since it would be nice to write `it.onVersions(["7.0"])("can do this thing", function() {})`

Reply:
Something kinda relevant I found for extending Mocha's context: https://github.com/mochajs/mocha/wiki/Third-party-UIs

Reply:
I believe we could come up with our custom Mocha plugin that achieve the same goal.  Unfortunately it seems that I will have to find some time in the future to investigate that.  Is it fine to use current PR for an initial v0.1 version 😄

Reply:
Yeah I think you have a good start, at least enough to try it out for one package and collect feedback.